include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3687,6 +3687,247 @@ struct OrtApi {
   ORT_API2_STATUS(RegisterCustomOpsUsingFunction, _Inout_ OrtSessionOptions* options,
                   _In_ const char* registration_func_name);
 
+  /** \brief Logs a message at the given severity level using ::OrtEnv's logger.
+   *
+   * Must create an ::OrtEnv before using this function, otherwise an error status is returned.
+   * Only messages with a severity level equal or greater than the ::OrtEnv's logging severity level
+   * are logged.
+   *
+   * Can be used to log messages in custom operator libraries.
+   *
+   * \param[in] log_severity_level The message's severity level.
+   * \param[in] message The message to log.
+   * \param[in] file_path The filepath of the file in which the message is logged. Usually the value of __FILE__.
+   * \param[in] line_number The file line number in which the message is logged. Usually the value of __LINE__.
+   * \param[in] func_name The name of the function in which the message is logged. Usually the value of __FUNCTION__.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   * \since Version 1.14
+   */
+  ORT_API2_STATUS(LogMessage, OrtLoggingLevel log_severity_level, _In_z_ const char* message, _In_z_ const char* file_path,
+                  int line_number, _In_z_ const char* func_name);
+
+  /** \brief Get the logging severity level of the ::OrtEnv's logger.
+   *
+   * Must create an ::OrtEnv before using this function, otherwise an error status is returned.
+   *
+   * \param[out] out Pointer to variable assigned with the logging severity level on success.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   * \since Version 1.14
+   */
+  ORT_API2_STATUS(GetLoggingSeverityLevel, _Out_ OrtLoggingLevel* out);
```

Comment:
Should this and the above `LogMessage` take in an `OrtEnv*`? 
Pros:
- Consistency since these functions operate on the env
- Less likely for users to call without having created an `OrtEnv` first.

Cons:
- Would have to create an API for retrieving the default `OrtEnv` for custom ops.

@yuslepukhin